### PR TITLE
[SaFeRDialogues] dataset and model release

### PR DESCRIPTION
**Pull request by meganung**
**Patch description**
adding the SaFeRDialogues dataset `-t saferdialogues` and model `-mf zoo:saferdialogues/model`

Updated the task list and model zoo list. Added the license in the task folder.

**Testing steps**
`pytest parlai/tasks/saferdialogues/test.py ` passes! 
`parlai em -t saferdialogues -mf zoo:saferdialogues/model`

### CODE REVIEW

File: parlai/tasks/saferdialogues/agents.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        opt = copy.deepcopy(opt)
+        opt['parlaidialogteacher_datafile'] = _path(opt)
+        super().__init__(opt, shared)
+
+    def _setup_data(self, path):
+        super()._setup_data(path)
+        if not self.opt['recovery']:
+            for i, ep in enumerate(self.episodes):
+                # make the signaling msg the label and remove the recovery msg
+                texts = ep[0]['text'].split('\n')
```
Comment (jxmsML) [inline parlai/tasks/saferdialogues/agents.py:50]: is it that all the examples are flattened?
i.e. one example per episode?

File: projects/saferdialogues/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+We collect a dataset of 8k dialogues demonstrating safety failures, feedback signaling them, and a response acknowledging the feedback. We show how fine-tuning on this dataset results in conversations that human raters deem considerably more likely to lead to a civil conversation, without sacrificing engagingness or general conversational ability.
+
+## Paper
+
+[Link](https://arxiv.org/abs/2110.07518)
+
+## Data
+
+```
+parlai display_data -t light_genderation_bias
```
Comment (jxmsML) [inline projects/saferdialogues/README.md:17]: hmm typo? -t saferdialogues?

**Review by jxmsML — COMMENTED** [review]
LGTM!! Thanks for the OSS efforts! some minor comments

File: parlai/tasks/saferdialogues/agents.py (hunk 1 above)
Comment (meganung) [reply parlai/tasks/saferdialogues/agents.py:50]: yes it's flattened. Added a note in the readme!

File: projects/saferdialogues/README.md (hunk 2 above)
Comment (meganung) [reply projects/saferdialogues/README.md]: good catch !! thanks

File: parlai/tasks/saferdialogues/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Dataset has been released under the CC BY-NC license. Please refer to the LICENSE file in this folder for more information.
+
+[ArXiv Paper](https://arxiv.org/abs/2110.07518)
+
+## SaferDialoguesTeacher
+Returns examples like so: 
+- [text]:  flattened context with the feedback signaling message as the last line in the context 
+- [labels]: recovery response acknowledging the feedback 
+
+Note: The dataset is flattened, so there is one episode per example.
```
Comment (jxmsML) [inline parlai/tasks/saferdialogues/README.md:15]: nit: "one example per episode"

**Review by stephenroller — APPROVED** [review]
Looks very complete to me.

**Pull request merged**